src/pages/index.js

```diff
@@ -2,51 +2,137 @@ import "./index.css";
 import Card from "../components/Card.js";
 import FormValidator from "../components/FormValidator.js";
 import Section from "../components/Section.js";
-import initialCards from "../utils/constants.js";
 import PopupWithImage from "../components/PopupWithImage.js";
 import PopupWithForm from "../components/PopupWithForm.js";
 import UserInfo from "../components/UserInfo.js";
+import Api from "../components/Api.js";
+import PopupDeleteCard from "../components/PopupDeleteCard.js";
 
 //Buttons
 const profileEditButton = document.querySelector("#profile-edit-btn");
 const addButton = document.querySelector("#add-button");
+const editAvatarButton = document.querySelector(".avatar-edit-button");
 
-function addCard (cardDataObj) {
-  const cardInstance = new Card(cardDataObj, "#card-template", (event) => {
-    imageModalHandler.open(event);
-  });
-  
-  const domCard = cardInstance.returnCard();
-
-  gridHandler.addItem(domCard);
-}
+//modalHandlers
+const deleteModalHandler = new PopupDeleteCard("#delete-card-modal");
+deleteModalHandler.setEventListeners();
 
-const profileInfoSelectors = {
-  nameSelector: ".profile__user-name",
-  jobSelector: ".profile__job"
-}
-
-const profileInfo = new UserInfo(profileInfoSelectors);
-
-//Modal Handlers
 const imageModalHandler = new PopupWithImage("#image-modal");
 imageModalHandler.setEventListeners();
 
 const profileModalHandler = new PopupWithForm("#profile-edit-modal", (data) => {
-  profileInfo.setUserInfo(data)
+  profileModalHandler.changeButtonText(true);
+  api.updateProfileInfo(data)
+  .then(json => {
+    profileInfo.setUserInfo(json);
+  })
+  .catch(err => {
+    console.error(err);
+  })
+  .finally(() => {
+    profileModalHandler.changeButtonText(false);
+  })
 });
 profileModalHandler.setEventListeners();
 
-const cardModalHandler = new PopupWithForm("#add-card-modal", addCard);
+const avatarModalHandler = new PopupWithForm("#avatar-edit-modal", (data) => {
+  avatarModalHandler.changeButtonText(true);
+  api.updateAvatar(data)
+  .then(json => {
+    api.updateAvatar(json);
+    profileInfo.setAvatar(json);
+  })
+  .catch(err => {
+    console.error(err);
+  })
+  .finally(() => {
+    avatarModalHandler.changeButtonText(false);
+  })
+});
+avatarModalHandler.setEventListeners();
+
+const cardModalHandler = new PopupWithForm("#add-card-modal",
+  data => {
+    cardModalHandler.changeButtonText(true);
+    api.addCard(data)
+    .then(json => {
+      addCard(json);
+    })
+    .catch(err => {
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors